src/components/theme-toggle-button.js

```diff
@@ -0,0 +1,35 @@
+// Needed for the CSS.
+import icon from './theme-toggle-button.png'
+
+const DARK = '(prefers-color-scheme: dark)'
+const DARKCLASS = 'dark-theme'
+
+export default function ThemeToggleButton() {
+	// The <html> element.
+	const el = document.documentElement
+
+	// Set class if user prefers darkness.
+	if (window.matchMedia(DARK).matches) {
+		el.classList.add(DARKCLASS)
```

Comment:
think react would like us no to manipulate classNames like this anywhere in the app.
 One thing we can use, would be close to an `ember.service`, a `react.context`.
 https://reactjs.org/docs/context.html
 That is, if we want anything to do with, for example a `UserStyles` (?) context, which could give some data to lower components, without having to pass all the data down through each parent/child (by connecting to the `react.context.user_styles`

Reply:
True makes sense.

Right now it relies on the `<html>` element. Could it target the top level react element instead?

Reply:
Yes! The link to the context doc is implementing dark mode as an example :}

Reply:
Haha, the guide even has a theme toggle button!

We can try that if/when we need access to the "theme" inside javascript land? At the moment it seems fine with just CSS land.